include/pagmo_plugins_nonfree/worhp.hpp

```diff
@@ -0,0 +1,1203 @@
+#ifndef PAGMO_WORHP_HPP
+#define PAGMO_WORHP_HPP
+
+#include <algorithm> // std::min_element, std::sort, std::remove_if
+#include <boost/dll/import.hpp>
+#include <boost/dll/shared_library.hpp>
+#include <boost/filesystem.hpp>
+#include <iomanip>
+#include <mutex>
+#include <pagmo/algorithm.hpp>
+#include <pagmo/algorithms/not_population_based.hpp>
+#include <pagmo/config.hpp>
+#include <pagmo/exceptions.hpp>
+#include <pagmo/io.hpp>
+#include <pagmo/population.hpp>
+#include <pagmo/problem.hpp>
+#include <pagmo/utils/constrained.hpp>
+#include <random>
+#include <stdexcept>
+#include <string>
+#include <tuple>
+#include <type_traits> // std::false_type
+#include <unordered_map>
+#include <vector>
+
+extern "C" {
+#include "bogus_libs/worhp_lib/worhp.h"
```

Comment:
That include is in the bogus_library build. I considered this to be "outisde" the ecosystem as the user will never see nor care about it.

I tried to add in CMAKE
```
include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include/pagmo_plugins_nonfree/bogus_libs/worhp_lib)
add_library(worhp_c SHARED ../include/pagmo_plugins_nonfree/bogus_libs/worhp_lib/worhp_bogus.c)
target_link_libraries(worhp_c pagmo_plugins_nonfree)
```
but it would still not pick it up. I can investigate further if you think it has to be done.